### DISCUSSION TITLE
How in admin panel, can i $get inside textinput ?

### DISCUSSION THREAD

**Discussion by yamakrane** [filamentphp/filament#5839] [Help]
Hi,

I have an invoice builder using a repeater for different products. Knowing that every product has price, I have to calculate the sum of the product prices and show them in a textinput named 'total_price' that i could send with form $data.

I have put that using $set closure, but it is showing too many conditions. Isn't there a way, where I can use $get closure inside my textinput ? Maybe something like:

```
Forms\Components\TextInput::make('total_price')
                    ->formatStateUsing(function (Closure $get){
                        $coupon = Coupon::find($get('coupon_id'));

                        if($coupon){
                            $discount = $coupon->reduction/100;
                        } else {
                            $discount = 0;
                        }

                        $total = collect($get('orderProducts'))->pluck('price')->sum();

                        return $total - $total * $discount;

                    })

```

The problem here is that ->formatStateUsing doesn''t reformat again the textinput everytime i add a product or change it.

Please help.

**Comment by danharrin** [comment]
`formatStateUsing()` only runs when the form field loads, you probably want to `$set` this field from `afterStateUpdated()` when it needs to change

**Reply by yamakrane** [reply]
When you use a repeater and then set an input outside that repeater and then delete an item, the item doesn’t get updated because the textinput is already set and needs to be reset on delete the item again … anyways, that’s just one thing. We really need a $get closure for input. Idk know how to do that ->placeholder() works like a charm.
